webv2/README.md

```diff
@@ -0,0 +1,504 @@
+    }
+  ],
+  "varchar": [
+    {
+      "T": "BYTES",
+      "Brief": ""
+    },
+    {
+      "T": "STRING",
+      "Brief": ""
+    }
+  ]
+}
+```
+
+(2) `/typemap/global` is a POST API to customize schema conversion on global level.
+The request body contains a map from source type to Spanner type in json format.
+
+#### Method
+
+`POST`
+
+#### Request body
+
+Map from source type to Spanner type.
+
+Example
+
+```json
+{
+  "smallint": "STRING",
+  "varchar": "BYTES"
+}
+```
+
+#### Response body
+
+Updated Conv struct in JSON format.
+
+(3) `/typemap/table?table=<table_name>` is a POST API which performs following
+operations on a single table.
+
+- Remove column
+- Rename column
+- Remove or Add primary key
+- Update type of column
+- Remove or Add NOT NULL constraint
+
+#### Method
+
+`POST`
+
+#### Request body
+
+Column wise actions to be performed with following possible values for each field.
+
+- Removed : true/false
+- Rename : New name or empty string
+- PK : "" | "ADDED" | "REMOVED"
+- NotNull : "" | "ADDED" | "REMOVED"
+- ToType : New Spanner type or empty string
+
+Example
+
+```json
+{
+  "UpdateCols": {
+    "AlbumTitle": {
+      "Removed": false,
+      "Rename": "AlbumName",
+      "PK": "",
+      "NotNull": "ADDED",
+      "ToType": "BYTES"
+    }
+  }
+}
+```
+
+#### Response body
+
+Updated Conv struct in JSON format.
+
+### Report file
+
+`/report` is a GET API which generates report file and returns file path.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+Path to report file.
+
+Example
+
+```
+/path/to/reportfile.report.txt
+```
+
+### Schema file
+
+`/schema` is a GET API which generates schema file and returns file path.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+Path to schema file.
+
+Example
+
+```
+/path/to/schemafile.schema.txt
+```
+
+### Interleave tables
+
+`/setparent?table=<table_name>` is a GET API which checks whether it is
+possible to convert a table into a Spanner interleaved table. If this conversion is possible,
+then the schema is changed and the parent table name is returned.
+If the conversion is not possible, a failure message is returned.
+
+#### Method
+
+`GET`
+
+#### Request body
+
+No request body is needed.
+
+#### Response body
+
+If it is possible to make given table interleaved, it will return with table
+name of parent.
+
+Example
+
+```json
+{
+  "Possible": true,
+  "Parent": "Singers",
+  "Comment": ""
+}
+```
+
+If it is not possible to interleave given table, it will return with the reason
+of why it failed.
+
+Example
+
+```json
+{
+  "Possible": false,
+  "Parent": "",
+  "Comment": "No valid prefix"
+}
+```
+
+### Drop foreign key
+
+`/drop/fk?table=<table_name>&pos=<position>` is a GET API which takes table name
+and array position of foreign key in query params. It drops foreign key at given
+position for given table name.
+
+#### Method
+
+`GET`
```

Comment:
Any reason we chose GET /drop/fk over DELETE /fk ?

Reply:
We have not updated the existing FK apis yet. We have one item in the backlog list for readme.